src/project.js

```diff
@@ -0,0 +1,134 @@
+};
+
+const description = (task, selectedProject) => {
+  const selectedTaskId1 = task.id;
+  localStorage.setItem(LOCAL_STORAGE_SELECTED_TASK, selectedTaskId1);
+  renderdescription(selectedProject);
+  return selectedTaskId1;
+};
+
+const rendertask = () => {
+  taskContainer.innerHTML = '';
+  descriptionContainer.innerHTML = '';
+  deleteTask.style.visibility = 'hidden';
+  updateTask.style.visibility = 'hidden';
+  const selectedProjectId1 = localStorage.getItem(LOCAL_STORAGE_SELECTED_PROJECT);
+  const selectedProject = projects.find((project) => project.id === selectedProjectId1);
+  taskContainer.innerHTML = `Project Name:-${selectedProject.name}`;
+  selectedProject.todos.forEach((task) => {
+    const taskElement = document.createElement('li');
+    taskElement.dataset.taskId = task.id;
+    taskElement.innerText = task.name;
+    taskElement.onclick = () => { description(task, selectedProject); };
+    taskContainer.appendChild(taskElement);
+  });
+};
+
+const task = (project) => {
+  addTask.style.visibility = 'visible';
+  const selectedProjectId1 = project.id;
+  localStorage.setItem(LOCAL_STORAGE_SELECTED_PROJECT, selectedProjectId1);
+  rendertask();
+};
+
+const renderProjects = () => {
+  projectsContainer.innerHTML = '';
+  descriptionContainer.innerHTML = '';
+  deleteTask.style.visibility = 'hidden';
+  updateTask.style.visibility = 'hidden';
+  projects.forEach((project) => {
+    const projectElement = document.createElement('li');
+    projectElement.dataset.projectId = project.id;
+    projectElement.innerText = project.name;
+    projectElement.onclick = () => { task(project); };
+    projectsContainer.appendChild(projectElement);
+  });
+};
+
+const save = () => {
+  localStorage.setItem(LOCAL_STORAGE_ALL_PROJECTS, JSON.stringify(projects));
+};
+
+const display = () => {
+  save();
+  renderProjects();
+};
+
+deleteTask.addEventListener('click', () => {
+  const selectedProjectId1 = localStorage.getItem(LOCAL_STORAGE_SELECTED_PROJECT);
+  const selectedProject = projects.find((project) => project.id === selectedProjectId1);
+  const selectedTaskId1 = localStorage.getItem(LOCAL_STORAGE_SELECTED_TASK);
+  const selectedTask = selectedProject.todos.find((task) => task.id === selectedTaskId1);
+  const Index = selectedProject.todos.indexOf(selectedTask);
+  selectedProject.todos.splice(Index, 1);
+  save();
+  rendertask();
+});
+
+updateTask.addEventListener('click', () => {
+  const selectedProjectId1 = localStorage.getItem(LOCAL_STORAGE_SELECTED_PROJECT);
+  const selectedProject = projects.find((project) => project.id === selectedProjectId1);
+  const selectedTaskId1 = localStorage.getItem(LOCAL_STORAGE_SELECTED_TASK);
+  const selectedTask = selectedProject.todos.find((task) => task.id === selectedTaskId1);
+  descriptionContainer.innerHTML = '';
+  deleteTask.style.visibility = 'hidden';
+  updateTask.style.visibility = 'hidden';
+
+  taskFormBody.style.visibility = 'visible';
```

Comment:
- [ ] [REQUIRED] 
Keep all the styling logic in CSS files (If you need to change something like a background image, add the classes with the different backgrounds to the CSS and then change the class name attribute with JavaScript) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/styling_requirement.png) from Pathwright)